### PR TITLE
Added more generic search method decoupled from TAO

### PR DESCRIPTION
**Pull request by gabrielfs7**
Added more generic search method decoupled from TAO required to do specific searches

Related PRs:

- [ ] https://github.com/oat-sa/lib-tao-elasticsearch/pull/43
- [ ] https://github.com/oat-sa/tao-core/pull/2790
- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/4

https://oat-sa.atlassian.net/browse/CON-497

### CODE REVIEW

**Review by bartlomiejmarszal — COMMENTED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by bartlomiejmarszal — APPROVED** [review]
Tested on my local machine

**Review by pnal — APPROVED** [review]
LGTM
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: src/ElasticSearch.php [hunk 1]
```diff
@@ -37,14 +37,43 @@
...
+                [
+                    'query' => [
+                        'query_string' => [
+                            "default_operator" => "AND",
+                            "query" => $query->getQueryString()
+                        ]
+                    ],
+                    "size" => $query->getLimit(),
+                    "from" => $query->getOffset(),
+                    "sort" => [],
```
Comment (shpran) [inline src/ElasticSearch.php:62]:
```suggestion
                            'default_operator' => 'AND',
                            'query' => $query->getQueryString(),
                        ]
                    ],
                    'size' => $query->getLimit(),
                    'from' => $query->getOffset(),
                    'sort' => [],
```

File: test/ElasticSearchTest.php [hunk 2]
```diff
@@ -116,6 +118,38 @@ protected function setUp(): void
...
+                        'query_string' => [
+                            "default_operator" => "AND",
+                            "query" => 'a:"b"'
+                        ]
+                    ],
+                    "size" => 777,
+                    "from" => 7,
+                    "sort" => [],
+                ]
+            )
```
Comment (shpran) [inline test/ElasticSearchTest.php:137]:
```suggestion
                            'default_operator' => 'AND',
                            'query' => 'a:"b"',
                        ]
                    ],
                    'size' => 777,
                    'from' => 7,
                    'sort' => [],
                ],
            ),
```

**Review by shpran — APPROVED** [review]
👍🏻 

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: test/ElasticSearchTest.php (hunk 2 above)
Comment (gabrielfs7) [reply test/ElasticSearchTest.php]: Done

File: src/ElasticSearch.php (hunk 1 above)
Comment (gabrielfs7) [reply src/ElasticSearch.php]: Done

**Pull request merged**